source/Irrlicht/CIrrDeviceStub.cpp

```diff
@@ -48,7 +48,7 @@ CIrrDeviceStub::CIrrDeviceStub(const SIrrlichtCreationParameters& params)
 
 	checkVersion(params.SDK_version_do_not_use);
 
-    IncludeHandler = core::make_smart_refctd_ptr<asset::IIncludeHandler>(new asset::CIncludeHandler(FileSystem), core::dont_grab);
+    IncludeHandler = new asset::CIncludeHandler(FileSystem);

```

Comment:
why ditch the smart_ptr ?

Reply:
gives compile errors with abstract classes

Reply:
has the same syntax as 
https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique
you don't need the new, just the args

Reply:
`IncludeHandler = core::make_smart_refctd_ptr<asset::CIncludeHandler>(FileSystem);

Reply:
ah, my bad, it was because i wrote core::make_smart_refctd_ptr<asset:: **IIncludeHandler** >

Reply:
But `core::make_smart_refctd_ptr<asset::CIncludeHandler>(FileSystem, core::dont_grab);` doesnt work, it wants to put dont_grab into CIncludeHandler's constructor :/

Reply:
you don't need a `core::dont_grab` I don't think so

Reply:
if you do then I've fucked up the definition of `template <class T> ...... core::make_smart_refctd_ptr`

Reply:
>you don't need a core::dont_grab I don't think so

well, it's easy to workaround (just drop after creating smart ptr), but there's
```
	template< class T, class... Args >
	inline smart_refctd_ptr<T> make_smart_refctd_ptr(Args&& ... args, dont_grab_t t)
```
so seems like it's supposed to work. But I don't know why it doesn't or how to do it other way

`dont_grab` is needed because while creating a ref-counted object it obviously has ref-count=1 and after creating smart ptr, it then has ref-count=2. But smart pointer's dtor drops only once so it's a memleak

Reply:
`make_smart_refctd_ptr` should just `dont_grab` by default

Reply:
I fucked up bigtime, the only definition of `make_smart_refctd_ptr` should have been:

```cpp
template< class T, class... Args >
	inline smart_refctd_ptr<T> make_smart_refctd_ptr(Args&& ... args)
	{
		T* obj = new T(std::forward<Args>(args)...);
		smart_refctd_ptr<T> smart(obj,dont_grab);
		return smart;
}
```

Reply:
ah, ok, so i'll push it to this branch